src/ImageSharp/Formats/Pbm/BinaryDecoder.cs

```diff
@@ -71,7 +71,11 @@ private static void ProcessGrayscale<TPixel>(Configuration configuration, Buffer
 
         for (int y = 0; y < height; y++)
         {
-            stream.Read(rowSpan);
+            if (stream.Read(rowSpan) == 0)
```

Comment:
This is lame actually. I should have sliced down `rowSpan` with the result of `stream.Read()`, for the pixel conversion done later, or used the condition `stream.Read(rowSpan) < rowSpan.Length` (the latter isstricter, but simpler).

This has little practical relevance (`BuffferedReadStream` is hitting EOF twice & `FromL8Bytes` is converting some memory garbage for broken files), so no need to fix it now, but the code looks stupid :)

Reply:
We can improve it with follow up.

Reply:
#2552